static/js/actions/ui.js

```diff
@@ -1,82 +1,85 @@
+// @flow
 // general UI actions
+import type { Action } from '../flow/generalTypes';
+
 export const CLEAR_UI = 'CLEAR_UI';
 export const clearUI = () => ({ type: CLEAR_UI });
 
 export const UPDATE_DIALOG_TEXT = 'UPDATE_DIALOG_TEXT';
-export const updateDialogText = text => (
+export const updateDialogText = (text: string): Action => (
   { type: UPDATE_DIALOG_TEXT, payload: text }
 );
 
 export const UPDATE_DIALOG_TITLE = 'UPDATE_DIALOG_TITLE';
-export const updateDialogTitle = title => (
+export const updateDialogTitle = (title: string): Action => (
   { type: UPDATE_DIALOG_TITLE, payload: title }
 );
 
 export const SET_DIALOG_VISIBILITY = 'SET_DIALOG_VISIBILITY';
-export const setDialogVisibility = bool => (
+export const setDialogVisibility = (bool: boolean): Action => (
   { type: SET_DIALOG_VISIBILITY, payload: bool }
 );
 
 // work history actions
 export const SET_WORK_HISTORY_EDIT = 'SET_WORK_HISTORY_EDIT';
-export const setWorkHistoryEdit = bool => (
+export const setWorkHistoryEdit = (bool: boolean): Action => (
   { type: SET_WORK_HISTORY_EDIT, payload: bool }
 );
 
 export const SET_WORK_DIALOG_VISIBILITY = 'SET_WORK_DIALOG_VISIBILITY';
-export const setWorkDialogVisibility = bool => (
+export const setWorkDialogVisibility = (bool: boolean): Action => (
   { type: SET_WORK_DIALOG_VISIBILITY, payload: bool }
 );
 
 export const SET_WORK_DIALOG_INDEX = 'SET_WORK_DIALOG_INDEX';
-export const setWorkDialogIndex = index => (
+export const setWorkDialogIndex = (index: number): Action => (
   { type: SET_WORK_DIALOG_INDEX, payload: index }
 );
 
 // dashboard actions
 export const TOGGLE_DASHBOARD_EXPANDER = 'TOGGLE_DASHBOARD_EXPANDER';
-export const toggleDashboardExpander = (courseId, newValue) => ({
+export const toggleDashboardExpander = (courseId: number, newValue: any): Action => ({
```

Comment:
there were a couple places where I had trouble tracking down exactly what type something should be - usually I tried to narrow it down (e.g. `string|number` or something), so if you spot something where we should be more specific that's great. I think we want as few things with type `any` as possible.